### PR TITLE
[MTS][87766816] Compile .js to dist/ directory

### PR DESCRIPTION
**Pull request by michaelstalker**
https://www.pivotaltracker.com/story/show/87766816
https://www.pivotaltracker.com/story/show/88072428

This will make the library more consistent with other JavaScript libraries at the company.

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -9,13 +9,13 @@ var del = require('del');
 gulp.task('dist', function() {
   gulp.src([ 'src/**/*.coffee' ])
   .pipe(coffee({bare: true}).on('error', gutil.log))
-  .pipe(gulp.dest('./'));
+  .pipe(gulp.dest('./dist'));
 });
 
 gulp.task('coffee', function() {
```
Comment (warrenv) [inline gulpfile.js]: Other than an argument difference to coffee() - this task is the same as diff. Could one be eliminated?

Comment (michaelstalker) [inline gulpfile.js]: Good catch. Yes, let's delete one.
1. Do we want the `bare` option set to `true`? Your [templates](https://github.com/rentpath/generator-rentpath-flight/blob/dev/app/templates/gulpfile.js) have that set to `true`, which compiles the scripts without a top-level function wrapper.
2. Do we want to use the `header` option, which prepends `// Generated by CoffeeScript 1.9.0` to the top of each file. This is useful for development, but not so useful in QA and production. Maybe AG's build process strips out these comments in production. Do you know if it does?

Comment (c0) [inline gulpfile.js]: Comments are stripped out.

**Comment by warrenv** [comment]
Please update the README to include instructions for running 'npm install' and 'bower install'.

**Comment by warrenv** [comment]
The tests are not running for me:

it-mbp-wvosper:map.js wvosper$ karma start
INFO [karma]: Karma v0.12.31 server started at http://localhost:9876/
INFO [launcher]: Starting browser Chrome
WARN [watcher]: Pattern "/Users/wvosper/source/map.js/app/bower_components/jquery/dist/jquery.js" does not match any file.
INFO [Chrome 40.0.2214 (Mac OS X 10.10.2)]: Connected on socket FDI5ehmuFU8KmDr8rG0M with id 95695259
Chrome 40.0.2214 (Mac OS X 10.10.2) ERROR: 'There is no timestamp for /base/app/bower_components/jquery/jquery.js!'

WARN [web-server]: 404: /base/app/bower_components/jquery/jquery.js
Chrome 40.0.2214 (Mac OS X 10.10.2) ERROR
  Uncaught Error: Script error for: jquery
  http://requirejs.org/docs/errors.html#scripterror
  at /Users/wvosper/source/map.js/node_modules/requirejs/require.js:141

**Comment by warrenv** [comment]
What do you think about adding the 'scripts' block from package.json in gallery.js?

**Comment by michaelstalker** [comment]
@warrenv I updated the README, fixed the test, and added the `scripts` block from `package.json` in gallery.js. I removed some scripts that weren't relevant map.js, like deleting the `.tmp/` directory. I included much of the README information from gallery.js.

**Comment by warrenv** [comment]
:+1:  Very nice work - removing the need to compile coffeescript to .tmp is a nice feature that we can probably adopt for the other js repos.

**Pull request merged**